### ISSUE TITLE
Fine-tune Citrinet config file

### ISSUE THREAD

**Issue by Omarnabk** [NVIDIA/NeMo#2405]
I am following tutorial "08_ASR_with_Subword_Tokenization.ipynb" to fine-tune Citrinet. 

My question is: which configuration file should I modify for fine-tuning: 
option 1: the original config file of the model obtained from **asr_model.cfg** 
option 2: download **config_bpe** [here ](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/citrinet/config_bpe.yaml)

Note, I want to use "**stt_en_conformer_ctc_large**"  model. 

Thanks in advance!

**Comment by titu1994** [comment]
stt_en_conformer_ctc_large as it's name suggests is Conformer model - not Citrinet. 

You should use the config in the instantiated model itself - therefore asr_model.cfg and use that to prepare new datasets and tokenizer then fine tune

**Comment by Omarnabk** [comment]
Thanks for your reply. 
Can I use the default tokenizer? 
If I use my own tokenizer, should I replace the decoder vocabulary?

**Comment by titu1994** [comment]
If your language remains the same, and processing of text is same (English lower case alphabet, space and apostrophe) then yes it is fine to use the default tokenizer in the model.

**Comment by Omarnabk** [comment]
In my case, the language is the same, but I'm trying to customize the model to a specific domain. I have a problem in transcribing non-English people's names/ domain-specific terms acnonyms and I was thinking of adding the names to the tokenizer vocabulary. Any recommendation?

**Comment by titu1994** [comment]
I think that would be easier with a domain specific language model than retraining the acoustic model. These models have been trained on several thousand hours of speech, so a language model might do the trick.

**Comment by Omarnabk** [comment]
I agree with you. I noticed that commercial ASR models ask for people names/ specific terms in advance, maybe to fine-tune their language model without touching the acoustic model. However, in my case, I am going to use the model for **non-native English Speakers** to transcribe their meetings. The pure model without fine-tuning obtained ~30%WER while fine-tuned on 10H Quatrznet obtained ~22%WER. 
I expect Citrinet or conformer_ctc_large to obtained better WER.

**Comment by Omarnabk** [comment]
I fine-tuned the stt_en_citrinet_256 model by changing the training set/ validation set/ and the LR in the model configuration file. However, when I load the generated checking point I get the following error: 

_FileNotFoundError: src path does not exist or it is not a path in nemo file. src value I got was: ../../manifests/asrset_1.4/tokenizers/no_appen/tokenizer_spe_unigram_v1024/tokenizer.model. Absolute: /home/manifests/asrset_1.4/tokenizers/no_appen/tokenizer_spe_unigram_v1024/tokenizer.model_

**Comment by titu1994** [comment]
The .ckpt does not have the Checkpoin tokenizer attached to it. You should use exp manager to save a Nemo model after every epoch so that it can easily be restored.

**Comment by Omarnabk** [comment]
@titu1994 Can you give an example of how to do it?
